meilisearch-http/tests/common.rs

```diff
@@ -1,15 +1,31 @@
 #![allow(dead_code)]
 
+use actix_web::{http::StatusCode, test};
 use serde_json::{json, Value};
 use std::time::Duration;
+use tempdir::TempDir;
+use tokio::time::delay_for;
 
-use actix_web::{http::StatusCode, test};
 use meilisearch_core::DatabaseOptions;
 use meilisearch_http::data::Data;
-use meilisearch_http::option::Opt;
 use meilisearch_http::helpers::NormalizePath;
-use tempdir::TempDir;
-use tokio::time::delay_for;
+use meilisearch_http::option::Opt;
+
+/// Performs a search test on both post and get routes
+#[macro_export]
+macro_rules! test_post_get_search {
+    ($server:expr, $query:expr, |$response:ident, $status_code:ident | $block:expr) => {
+        let post_query: meilisearch_http::routes::search::SearchQueryPost = serde_json::from_str(&$query.clone().to_string()).unwrap();
+        let get_query: meilisearch_http::routes::search::SearchQuery = post_query.into();
+        let get_query = ::serde_url_params::to_string(&get_query).unwrap();
+        let ($response, $status_code) = $server.search_get(&get_query).await;
+        let _ =::std::panic::catch_unwind(|| $block)
+            .map_err(|e| panic!("panic in get route: {:?}", e.downcast_ref::<&str>().unwrap()));
+        let ($response, $status_code) = $server.search_post($query).await;
+        let _ = ::std::panic::catch_unwind(|| $block)
+            .map_err(|e| panic!("panic in post route: {:?}", e.downcast_ref::<&str>().unwrap()));
+    };
+}
```

Comment:
Why do I see this macro in this PR?

Reply:
This was moved to `common` to be shared with the placeholder search tests :)